### ISSUE TITLE
bogus "Property 'fn' is missing" on every consumer

### ISSUE THREAD

**Issue by codingthat** [harshmandan/svelte-check-native#65]
Thanks for the quick turnaround on #56 and #57!  :)

With that unblocking the check on the rest of the same codebase, one class of false positives remains (also present in 1.5.2, so not a regression) — #58 also happened to fix a bunch of other faux errors I was seeing for the same codebase just applying temporary workarounds for #56, so that's a great bonus 😁 

So, the final frontier (for me, anyway) is `const { enhance, message } = superForm(...); export { enhance, message };` (sveltekit-superforms) and a `function validate()` exported for parents to call.  But it's reproducible purely / without sveltekit-superforms.

## Reproducible example

Along the lines of https://svelte.dev/docs/svelte/legacy-export-let#Renaming-props :

`src/Child.svelte`

```svelte
<script lang="ts">
export let label = '';
function validate() {
	return label.length > 0;
}
export { validate };
</script>

<span>{label}</span>
```

`src/Parent.svelte`

```svelte
<script lang="ts">
import Child from './Child.svelte';
</script>

<Child label="a" />
```

`tsconfig.json`

```json
{
  "compilerOptions": {
    "strict": true,
    "allowJs": true,
    "checkJs": true,
    "target": "esnext",
    "module": "esnext",
    "moduleResolution": "bundler",
    "verbatimModuleSyntax": true,
    "skipLibCheck": true,
    "types": []
  },
  "include": ["src/**/*.ts", "src/**/*.svelte"]
}
```

svelte 5.57.0, typescript 7.0.2 as the engine (svelte-check itself run against typescript 6.0.3).

svelte-check 4.3.6 and 4.7.6 are clean on this; svelte-check-native 1.5.2 and 1.6.0 error on every consumer of the component:

```
$ svelte-check --tsconfig ./tsconfig.json
COMPLETED … 0 ERRORS 0 WARNINGS

$ svelte-check-native --tsconfig ./tsconfig.json
src/Parent.svelte:5:2
Error: Property 'validate' is missing in type '{ label: string; }' but required in type '{ label?: string | undefined; validate: any; }'. (ts)
```

## Affected forms

| child pattern | svelte-check-native props type | consumer error |
|---|---|---|
| `function validate(){}` + `export { validate };` | `validate: any` (required) | yes |
| `function validate(){}` + `export { validate as check };` | `check: any` (required) | yes |
| `const { first, second } = obj;` + `export { first, second };` | `first: any; second: any` (required) | yes |
| `const helper = () => …;` + `export { helper };` | `helper?: typeof helper` | no |
| `export function validate(){}` (direct form) | `validate?: typeof validate` | no |

## Expected (svelte2tsx 0.7.61 for the first row)

```ts
return { props: {label: label , validate: validate} as {label?: typeof label, validate?: typeof validate}, exports: {} as any as { validate: typeof validate }, … }
```

## Actual (`--emit-ts`, first row)

```ts
return { props: undefined as any as ({ label?: typeof label; validate: any; }), …, exports: undefined as any as ({ label: typeof label; }) };
```

## Second consequence: the component API loses its type

`validate` is also dropped from the emitted `exports` type, so calling it through `bind:this` yields `any` and real errors go unreported:

```svelte
<script lang="ts">
import Child from './Child.svelte';
let child: Child;
function run() {
	const n: number = child.validate(); // validate() returns boolean
	return n;
}
</script>

<Child bind:this={child} label="a" />
<button on:click={run}>run</button>
```

```
$ svelte-check --tsconfig ./tsconfig.json
src/BindThisConsumer.svelte:5:8
Error: Type 'boolean' is not assignable to type 'number'. (ts)

$ svelte-check-native --tsconfig ./tsconfig.json
(no diagnostic on line 5; only the bogus "Property 'validate' is missing" on the <Child> tag)
```

So the same misclassification invents an error on every consumer and hides real errors on the component API.

----

Thanks again for this great project!  🎉 🎉 🎉

**Comment by harshmandan** [comment]
Thanks! Fixed in #66 and will be in the next release (1.7.0): a component's exported functions are now typed as instance members, so consumers no longer get the bogus "Property 'fn' is missing" error.
